src/DynamoCore/Models/RecordableCommands.cs

```diff
@@ -689,6 +689,110 @@ protected override void SerializeCore(XmlElement element)
             #endregion
         }
 
+        [DataContract]
+        public class AutoCreateNodeCommand : ModelBasedRecordableCommand
+        {
+            #region Public Class Methods
+
+            private void SetProperties(int outputPortIndex, int inputPortIndex, double x, double y, 
+                bool defaultPosition, bool transformCoordinates)
+            {
+                OutputPortIndex = outputPortIndex;
+                InputPortIndex = inputPortIndex;
+                X = x;
+                Y = y;
+                DefaultPosition = defaultPosition;
+                TransformCoordinates = transformCoordinates;
+            }
+
+            public AutoCreateNodeCommand(
+                NodeModel inputNode, NodeModel givenNode, int outputPortIndex, int inputPortIndex, 
+                double x, double y, bool defaultPosition, bool transformCoordinates)
+                : base(inputNode != null && givenNode != null ? new[] { inputNode.GUID, givenNode.GUID } : new[] { Guid.Empty })
```

Comment:
The `CreateNodeCommand` has a ctor that accepts a `NodeModel` and this new command is basically modeled after the `CreateNodeCommand` hence the reason for the `NodeModel` argument to be ok here. I'm not sure what you mean by `NodeModel` cannot be deserialzed? As mentioned in another comment, I don't see why this new command needs a `Deserialize` method?